packages/core/admin/server/src/services/encryption.ts

```diff
@@ -0,0 +1,69 @@
+import crypto from 'crypto';
+
+const IV_LENGTH = 16; // 16 bytes for AES-GCM IV
+
+const getHashedKey = (): Buffer | null => {
+  const rawKey: string = strapi.config.get('admin.secrets.encryptionKey');
+  if (!rawKey) {
+    strapi.log.warn('Encryption key is missing from config');
+    return null;
+  }
+
+  return crypto.createHash('sha256').update(rawKey).digest(); // Always 32 bytes
+};
+
+/**
+ * Encrypts a value string using AES-256-GCM.
+ * Returns a string containing IV, encrypted content, and auth tag (all hex-encoded).
+ */
+const encrypt = (value: string) => {
+  const key = getHashedKey();
+  if (!key) return null;
+  const iv = crypto.randomBytes(IV_LENGTH);
+  const cipher = crypto.createCipheriv('aes-256-gcm', key, iv);
+
+  let encrypted = cipher.update(value, 'utf8', 'hex');
+  encrypted += cipher.final('hex');
+
+  const authTag = cipher.getAuthTag();
+
+  return `${iv.toString('hex')}:${encrypted}:${authTag.toString('hex')}`;
```

Comment:
```suggestion
  return `v1:${iv.toString('hex')}:${encrypted}:${authTag.toString('hex')}`;
```
We may want to version this so we don't run into problems in the future if we have to change anything

Reply:
that's a very nice idea I like it, will add it